Middleware/src/test/java/ca/concordia/ngci/tools4cities/middleware/test/TestThreeCsvSum.java

```diff
@@ -0,0 +1,44 @@
+package ca.concordia.ngci.tools4cities.middleware.test;
+
+import java.util.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import ca.concordia.ngci.tools4cities.middleware.producers.ThreeCsvProducer;
+import ca.concordia.ngci.tools4cities.middleware.consumers.ThreeCsvSumConsumer;
+import ca.concordia.ngci.tools4cities.middleware.middleware.IProducer;
+
+public class TestThreeCsvSum {
+
+    @Test
+    public void testThreeCsvSum() {
+        // Create producers for three CSV files, each extracting values from the 4th column
+        final IProducer<String> producer1 = new ThreeCsvProducer("/home/zongo/Documents/ConcordiaLabs/Workspace/Tools4Cities-Middleware/Conso_Moy_RTA_SCIAN_CI.csv", null, 3);
```

Comment:
Please replace absolute paths with paths relative to the project. Whenever we push code that will go into main, we have to make sure it will be able to run on any computer, so you should not point to files that exist only on your computer.

I suggest you create mock data to replace the CSVs we cannot push into the repo. For example: https://docs.google.com/spreadsheets/d/1mY_-AJEo0TKVO6HVVOsBq8VaZ_THxKwfKfvdKMnDlmQ/edit?usp=drivesdk